src/objects.js

```diff
@@ -130,6 +146,70 @@ export class Page {
   mapData(func) {
     return new Page(this.data.map(func), this.before, this.after)
   }
+}
+
+/** Used to iterate over the pages of a query. */
+export class PageIterator {
+  /**
+   * @param {Client} client
+   * @param {number} opts.pageSize Number of elements in each page.
+   * @param opts.mapLambda
+   *   Mapping query (made by {@link lambda}) applied to each element of each page.
+   * @param {function} opts.map
+   *   Mapping JavaScript function used on each page element.
+   */
+  constructor(client, set, opts={}) {
+    this.client = client
+    this.set = set
+    /** Set to `true` when there are no elements left to page through. */
+    this.done = false
+
+    Object.assign(this, applyDefaults(opts, {
+      pageSize: null,
+      mapLambda: null,
+      map: null
+    }))
 
-  // TODO: setIterator is async, so need a different way to do that...
+    // Change nulls for pageSize to `undefined` because
+    // `undefined` is automatically removed from queries by JSON.stringify.
+    if (this.pageSize === null)
+      this.pageSize = undefined
+
+    this.direction = undefined
+    this.cursor = undefined
+  }
+
+  /**
+   * Moves this iterator to the next page and returns its data.
+   * May set `this.done` to `true`.
+   */
+  async next() {
```

Comment:
I'm not 100% sold on the current PageIterator API. 

It's currently in this weird half-async world, where the deferred methods end up manipulating the object's state, which will then be inspected synchronously. That seems weird to me, and, without async/await, I think the call sites will end up having to jump through hoops (recursion within a promise flatMap) in order to bridge the gap between the Promise monad and the iterative approach that having a `.done` boolean implies.

Without higher level constructs such as async/await, I think the API needs to be purely functional. In the end, it should probably behave more like a stream of pages, rather than an iterator; you should be able to apply a predicate to the stream (i.e. a `.takeWhile()`) method, and then you can act on the stream through a `.foreach` or a `.map`.

Let me know what you think.